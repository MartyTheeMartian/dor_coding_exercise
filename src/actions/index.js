import axios from 'axios';

//make a post request to retrieve to actual token
const retrieveToken = () => {
  const API_URL = 'http://api.getdor.com/v1/tokens';
  const body = {
    refresh_token: 'wSGgSqmSDKS4YPfMSHZ4YyiOpiNv'
  };
  let token = axios.post(API_URL, body).then((response) => {
    return response
  });

  return {type: 'RETRIEVE', payload: token}
}

//plug the token into the get request to retrieve the data back
const loadAllData = (token) => {
  const API_URL = 'http://api.getdor.com/v1/teams/4/stores/4/days';
  const config = {
    headers: {
      Authorization: `Bearer ${token}`
    }
  };
  let data = axios.get(API_URL, config).then((response) => {
    return response;
  });

  return {type: 'LOAD_ALL_DATA', payload: data}
}

//utilize thunk middleware to make sure the order of the actions is correct
//note: I am also using the promiseMiddleware here .
export const getAllDate = () => {
  return (dispatch, getState) => {
    dispatch(retrieveToken()).then(() => {
      const token = getState().retrieveToken;
      dispatch(loadAllData(token));
    })
  }
}
