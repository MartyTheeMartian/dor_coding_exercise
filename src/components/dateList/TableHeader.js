import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {getAllDate} from '../../actions';

const mapStateToProps = (state, ownProps) => {
  return {
    currentTime: state.currentTime.time,
  }
}

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    getAllDate
  },dispatch)
}

class TableHeader extends Component {

  render() {
    return (
      <thead>
        <tr>
          <th >
            <div id="refresh-wrapper" onClick={this.props.getAllDate}>
              <h4>
                <a id="refresh">
                  Refresh
                </a>
              </h4>
            </div>
          </th>
          <th>
            <h4>
              <strong>
                Last updated {this.props.currentTime}
              </strong>
            </h4>
          </th>
        </tr>
      </thead>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(TableHeader);
