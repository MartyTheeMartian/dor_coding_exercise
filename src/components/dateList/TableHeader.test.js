import React from 'react';
import { TableHeader, mapStateToProps, mapDispatchToProps } from './TableHeader';
import { shallow } from 'enzyme';
import appReducer from '../../reducers/index';
import toJson from 'enzyme-to-json'
import { getAllDate } from '../../actions';

describe('TableHeader Component', () => {

  it('should render without any props defined', () => {
    const component = shallow(
      <TableHeader />
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with props passed in', () => {
    const component = shallow(
      <TableHeader currentTime={'2017-09-09'} />
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('mapStateToProps', () => {
    const state = { currentTime: { time: '12-50-42' } };
    const expected = { currentTime: '12-50-42' };
    expect(mapStateToProps(state)).toEqual(expected);
  });

  it('simulate refresh clicked', () => {
    const wrapper = shallow(
      <TableHeader currentTime={'2017-09-09'}/>
    );
    expect(wrapper).toMatchSnapshot();

    wrapper.find('#refresh-wrapper').simulate('click');

    expect(wrapper).toMatchSnapshot();
  });

  it('matDispatchtoProps', () =>{
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch)).toHaveProperty('getAllDate');
  })
});
