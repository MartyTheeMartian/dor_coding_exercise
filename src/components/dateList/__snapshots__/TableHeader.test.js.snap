// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader Component should render with props passed in 1`] = `
<thead>
  <Header />
  <tr>
    <td
      id="refresh-wrapper"
    >
      <h4>
        <a
          id="refresh"
          onClick={[Function]}
        >
          Refresh
        </a>
      </h4>
    </td>
    <td
      id="update-wrapper"
    >
      <h4>
        <strong>
          Last updated 
          2017-09-09
        </strong>
      </h4>
    </td>
  </tr>
</thead>
`;

exports[`TableHeader Component should render without any props defined 1`] = `
<thead>
  <Header />
  <tr>
    <td
      id="refresh-wrapper"
    >
      <h4>
        <a
          id="refresh"
          onClick={[Function]}
        >
          Refresh
        </a>
      </h4>
    </td>
    <td
      id="update-wrapper"
    >
      <h4>
        <strong>
          Last updated 
        </strong>
      </h4>
    </td>
  </tr>
</thead>
`;

exports[`TableHeader Component simulate refresh clicked 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <thead>
    <Header />
    <tr>
        <td
            id="refresh-wrapper"
        >
            <h4>
                <a
                    id="refresh"
                    onClick={[Function]}
                >
                    Refresh
                </a>
            </h4>
        </td>
        <td
            id="update-wrapper"
        >
            <h4>
                <strong>
                    Last updated 
                    2017-09-09
                </strong>
            </h4>
        </td>
    </tr>
</thead>,
  "nodes": Array [
    <thead>
      <Header />
      <tr>
            <td
                  id="refresh-wrapper"
            >
                  <h4>
                        <a
                              id="refresh"
                              onClick={[Function]}
                        >
                              Refresh
                        </a>
                  </h4>
            </td>
            <td
                  id="update-wrapper"
            >
                  <h4>
                        <strong>
                              Last updated 
                              2017-09-09
                        </strong>
                  </h4>
            </td>
      </tr>
</thead>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        currentTime="2017-09-09"
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "currentTime": "2017-09-09",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <thead>
          <Header />
          <tr>
                    <td
                              id="refresh-wrapper"
                    >
                              <h4>
                                        <a
                                                  id="refresh"
                                                  onClick={[Function]}
                                        >
                                                  Refresh
                                        </a>
                              </h4>
                    </td>
                    <td
                              id="update-wrapper"
                    >
                              <h4>
                                        <strong>
                                                  Last updated 
                                                  2017-09-09
                                        </strong>
                              </h4>
                    </td>
          </tr>
</thead>,
        "_debugID": 6,
        "_renderedOutput": <thead>
          <Header />
          <tr>
                    <td
                              id="refresh-wrapper"
                    >
                              <h4>
                                        <a
                                                  id="refresh"
                                                  onClick={[Function]}
                                        >
                                                  Refresh
                                        </a>
                              </h4>
                    </td>
                    <td
                              id="update-wrapper"
                    >
                              <h4>
                                        <strong>
                                                  Last updated 
                                                  2017-09-09
                                        </strong>
                              </h4>
                    </td>
          </tr>
</thead>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    currentTime="2017-09-09"
/>,
}
`;

exports[`TableHeader Component simulate refresh clicked 2`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <thead>
    <Header />
    <tr>
        <td
            id="refresh-wrapper"
        >
            <h4>
                <a
                    id="refresh"
                    onClick={[Function]}
                >
                    Refresh
                </a>
            </h4>
        </td>
        <td
            id="update-wrapper"
        >
            <h4>
                <strong>
                    Last updated 
                    2017-09-09
                </strong>
            </h4>
        </td>
    </tr>
</thead>,
  "nodes": Array [
    <thead>
      <Header />
      <tr>
            <td
                  id="refresh-wrapper"
            >
                  <h4>
                        <a
                              id="refresh"
                              onClick={[Function]}
                        >
                              Refresh
                        </a>
                  </h4>
            </td>
            <td
                  id="update-wrapper"
            >
                  <h4>
                        <strong>
                              Last updated 
                              2017-09-09
                        </strong>
                  </h4>
            </td>
      </tr>
</thead>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        currentTime="2017-09-09"
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "currentTime": "2017-09-09",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <thead>
          <Header />
          <tr>
                    <td
                              id="refresh-wrapper"
                    >
                              <h4>
                                        <a
                                                  id="refresh"
                                                  onClick={[Function]}
                                        >
                                                  Refresh
                                        </a>
                              </h4>
                    </td>
                    <td
                              id="update-wrapper"
                    >
                              <h4>
                                        <strong>
                                                  Last updated 
                                                  2017-09-09
                                        </strong>
                              </h4>
                    </td>
          </tr>
</thead>,
        "_debugID": 6,
        "_renderedOutput": <thead>
          <Header />
          <tr>
                    <td
                              id="refresh-wrapper"
                    >
                              <h4>
                                        <a
                                                  id="refresh"
                                                  onClick={[Function]}
                                        >
                                                  Refresh
                                        </a>
                              </h4>
                    </td>
                    <td
                              id="update-wrapper"
                    >
                              <h4>
                                        <strong>
                                                  Last updated 
                                                  2017-09-09
                                        </strong>
                              </h4>
                    </td>
          </tr>
</thead>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    currentTime="2017-09-09"
/>,
}
`;
